hypervisor/dm/vpci/vpci.c

```diff
@@ -464,8 +464,12 @@ static int32_t write_pt_dev_cfg(struct pci_vdev *vdev, uint32_t offset,
 	} else if (sriovcap_access(vdev, offset)) {
 		write_sriov_cap_reg(vdev, offset, bytes, val);
 	} else {
-		/* passthru to physical device */
-		pci_pdev_write_cfg(vdev->pdev->bdf, offset, bytes, val);
+		/* For GVT-D, prevent stolen memory and opregion memory write */
+		if (!is_postlaunched_vm(vdev->vpci->vm) || is_gvtd(vdev->pdev->bdf) ||
+			((offset != PCIR_BDSM) && (offset != PCIR_ASLS_CTL))) {
```

Comment:
!(is_gvtd() && (offset ....))    ?
Just easier to read to me.

Reply:
refined